sequencer/src/bin/test-node.rs

```diff
@@ -0,0 +1,366 @@
+) -> io::Result<()> {
+    let mut app = tide_disco::App::<(), ServerError>::with_state(());
+    let toml_str = r#"
+[route.builder_address]
+PATH = ["builderaddress"]
+DOC = """
+Get the builder address.
+"""
+    "#;
+    let toml = toml::from_str::<toml::value::Value>(toml_str)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    let mut api = Api::<(), ServerError, Ver>::new(toml)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+    api.get("builder_address", move |_, _| {
+        async move { Ok(builder_address) }.boxed()
+    })
+    .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    app.register_module("block_info", api)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    spawn(app.serve(format!("0.0.0.0:{port}"), bind_version));
+    Ok(())
+}
+
+// Copied from `commitment_task::start_http_server`.
+// TODO: Remove these redundant code
+fn start_commitment_server<Ver: StaticVersionType + 'static>(
+    port: u16,
+    hotshot_address: Address,
+    bind_version: Ver,
+) -> io::Result<()> {
+    let mut app = tide_disco::App::<(), ServerError>::with_state(());
+    let toml = toml::from_str::<toml::value::Value>(include_str!("../../api/commitment_task.toml"))
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    let mut api = Api::<(), ServerError, Ver>::new(toml)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    api.get("gethotshotcontract", move |_, _| {
+        async move { Ok(hotshot_address) }.boxed()
+    })
+    .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    app.register_module("api", api)
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+    spawn(app.serve(format!("0.0.0.0:{port}"), bind_version));
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{process::Stdio, time::Duration};
+
+    use async_compatibility_layer::art::async_sleep;
+    use async_std::process::Command;
+    use escargot::CargoBuild;
+    use portpicker::pick_unused_port;
+    use reqwest::StatusCode;
+    use sequencer::{Header, Transaction};
+    use sequencer_utils::AnvilOptions;
+
+    // If this test failed and you are doing changes on the following stuff, please
+    // sync your changes to [`espresso-sequencer-go`](https://github.com/EspressoSystems/espresso-sequencer-go)
+    // and open a PR.
+    // - APIs update
+    // - Types (like `Header`) update
+    #[async_std::test]
+    async fn test_node_test() {
+        let anvil = AnvilOptions::default().spawn().await;
+        let builder_port = pick_unused_port().unwrap();
+        let commitment_task_port = pick_unused_port().unwrap();
+        let api_port = pick_unused_port().unwrap();
+        let postgres_port = pick_unused_port().unwrap();
+
+        let mut db = Command::new("docker")
+            .arg("compose")
+            .arg("up")
+            .arg("-d")
+            .arg("sequencer-db")
```

Comment:
We also need to add this to the binary don't we? I think ideally we don't use docker-compose for it but call docker directly so we don't need to be in this repo for it to work.

Reply:
This indeed makes it simpler to launch the test-node binary in sequencer repo. However, since developers would likely prefer using docker image directly, hardcoding the docker commands into the binary seems somewhat inappropriate.

Reply:
In https://github.com/EspressoSystems/espresso-sequencer/pull/1353/commits/1624cd465ebc03a0038f681f1b392a89344d07c9 we have a self contained docker image now that starts postgres, anvil and the dev node.